android/app/src/main/java/org/pathcheck/covidsafepaths/exposurenotifications/reactmodules/ExposureNotificationsModule.java

```diff
@@ -51,7 +51,7 @@ public void requestExposureNotificationAuthorization(final Promise promise) {
     FutureCallback<Void> callback = new FutureCallback<Void>() {
       @Override
       public void onSuccess(Void result) {
-        promise.resolve((Util.getEnStatusWritableArray(true)));
+        promise.resolve("Active");
```

Comment:
@juancruzgs , we ended up being able to change the api for the NativeModule function to be a just a string, rather than the tuple.  With `Active` being the value that indicates everything is on and `Disabled` would be the other string that could be return.

Is what we have here the correct way to implement this in the android side? If so how would be pass `Disabled` for the failed state, and would it make sense to remove the WritableArray stuff?

Reply:
Yes, this is correct.
To improve it you could add constants for "Active" and "Disabled" in `CallbackMessages.java`.

You also have to change `EventSender.sendExposureNotificationStatusChanged` to use those constants and then remove `getEnStatusWritableArray`